gen_moddeps.sh fixes: drop echo/printf from cycle && introduce xbasename() && fix including extra modules

reagentoo opened the pull request:
```
gen_moddeps.sh: don't use echo/printf inside the cycle

Also prevent printing empty line if mydeps column is null.

Signed-off-by: Dmitry Baranov <reagentoo@gmail.com>
```
```
gen_moddeps.sh: introduce xbasename()

Introduce xbasename() wrapper to use it instead of xargs basename.
It guards from two cases:
- zero count of module names is passing from pipe
- module name starting with "-" interprets as option

Signed-off-by: Dmitry Baranov <reagentoo@gmail.com>
```
```
gen_moddeps.sh: fix including extra modules when ALLRAMDISKMODULES="yes"

depmod util doesn't care about updating modules.order
when the new extra modules was installed.

Bug: https://bugs.gentoo.org/916233
Signed-off-by: Dmitry Baranov <reagentoo@gmail.com>
```

reagentoo committed:
gen_moddeps.sh: don't use echo/printf inside the cycle

Also prevent printing empty line if mydeps column is null.

Signed-off-by: Dmitry Baranov <reagentoo@gmail.com>

reagentoo committed:
gen_moddeps.sh: introduce xbasename()

Introduce xbasename() wrapper to use it instead of xargs basename.
It guards from two cases:
- zero count of module names is passing from pipe
- module name starting with "-" interprets as option

Signed-off-by: Dmitry Baranov <reagentoo@gmail.com>

reagentoo commented:
@FlyingWaffleDev @timkenhan
Hi. Please do your tests if possible.

timkenhan commented:
Not sure what to look for exactly, but I've tried building an initramfs with it. Boots without any issue so far.

reagentoo committed:
gen_moddeps.sh: fix including extra modules when ALLRAMDISKMODULES="yes"

depmod util doesn't care about updating modules.order
when the new extra modules was installed.

Bug: https://bugs.gentoo.org/916233
Signed-off-by: Dmitry Baranov <reagentoo@gmail.com>

thesamesam commented:
Thanks!
